src/gs-monitor.c

```diff
@@ -14,493 +14,441 @@
 
-        res = gs_listener_set_session_idle (monitor->priv->listener, is_idle);
+	res = gs_listener_set_session_idle(monitor->priv->listener, is_idle);
 
-        return res;
+	return res;
 }
 
-static gboolean
-release_grab_timeout (GSMonitor *monitor)
-{
-        gboolean manager_active;
+static gboolean release_grab_timeout(GSMonitor* monitor) {
+	gboolean manager_active;
 
-        manager_active = gs_manager_get_active (monitor->priv->manager);
-        if (! manager_active) {
-                gs_grab_release (monitor->priv->grab);
-        }
+	manager_active = gs_manager_get_active(monitor->priv->manager);
+	if (!manager_active) {
+		gs_grab_release(monitor->priv->grab);
+	}
 
-        monitor->priv->release_grab_id = 0;
-        return FALSE;
+	monitor->priv->release_grab_id = 0;
+	return FALSE;
 }
 
-static gboolean
-watcher_idle_notice_cb (GSWatcher *watcher,
-                        gboolean   in_effect,
-                        GSMonitor *monitor)
-{
-        gboolean activation_enabled;
-        gboolean handled;
-
-        gs_debug ("Idle notice signal detected: %d", in_effect);
-
-        /* only fade if screensaver can activate */
-        activation_enabled = gs_listener_get_activation_enabled (monitor->priv->listener);
-
-        handled = FALSE;
-        if (in_effect) {
-                if (activation_enabled) {
-                        /* start slow fade */
-                        if (gs_grab_grab_offscreen (monitor->priv->grab, FALSE)) {
-                                gs_fade_async (monitor->priv->fade, FADE_TIMEOUT, NULL, NULL);
-                        } else {
-                                gs_debug ("Could not grab the keyboard so not performing idle warning fade-out");
-                        }
-
-                        handled = TRUE;
-                }
-        } else {
-                gboolean manager_active;
-
-                manager_active = gs_manager_get_active (monitor->priv->manager);
-                /* cancel the fade unless manager was activated */
-                if (! manager_active) {
-                        gs_debug ("manager not active, performing fade cancellation");
-                        gs_fade_reset (monitor->priv->fade);
-
-                        /* don't release the grab immediately to prevent typing passwords into windows */
-                        if (monitor->priv->release_grab_id != 0) {
-                                g_source_remove (monitor->priv->release_grab_id);
-                        }
-                        monitor->priv->release_grab_id = g_timeout_add_seconds (1, (GSourceFunc)release_grab_timeout, monitor);
-                } else {
-                        gs_debug ("manager active, skipping fade cancellation");
-                }
-
-                handled = TRUE;
-        }
-
-        return handled;
+static gboolean watcher_idle_notice_cb(GSWatcher* watcher, gboolean in_effect,
+	GSMonitor* monitor) {
+	gboolean activation_enabled;
+	gboolean handled;
```

Comment:
Newline could probably be removed or given how many args there are, break this up.